cardano-testnet/test/cardano-testnet-test/Cardano/Testnet/Test/Gov/DRepActivity.hs

```diff
@@ -192,44 +192,25 @@ activityChangeProposalTest execConfig epochStateView configurationFile socketPat
   (EpochNo epochAfterProp) <- getCurrentEpochNo epochStateView
   H.note_ $ "Epoch after \"" <> prefix <> "\" prop: " <> show epochAfterProp
 
-  waitAndCheck epochAfterProp
+  mResult <- watchEpochStateView epochStateView isSuccess minWait maxWait
+
+  void $ H.evalMaybe mResult
 
   return thisProposal
 
   where
-    waitAndCheck :: (MonadTest m, MonadIO m)
-                 => Word64 -> m ()
-    waitAndCheck epochAfterProp = do
-      !eProposalResult
-        <- evalIO . runExceptT $ foldEpochState
-                                    (File configurationFile)
-                                    (File socketPath)
-                                    FullValidation
-                                    (EpochNo (epochAfterProp + maxWait))
-                                    ()
-                                    (\epochState _ _ -> filterEpochState (isSuccess epochAfterProp) epochState)
-      void $ evalEither eProposalResult
-
-    filterEpochState :: (EpochNo -> EpochInterval -> Bool) -> AnyNewEpochState -> StateT () IO LedgerStateCondition
-    filterEpochState f (AnyNewEpochState sbe newEpochState) =
-        caseShelleyToBabbageOrConwayEraOnwards
-          (const $ error "activityChangeProposalTest: Only conway era onwards supported")
-          (const $ do
-            let pParams = newEpochState ^. nesEpochStateL . epochStateGovStateL . curPParamsGovStateL . ppDRepActivityL
-                currEpoch = nesEL newEpochState
-            return (if f currEpoch pParams
-                    then ConditionMet
-                    else ConditionNotMet)
-          )
-          sbe
-
-    isSuccess :: Word64 -> EpochNo -> EpochInterval -> Bool
-    isSuccess epochAfterProp (EpochNo epochNo) (EpochInterval epochInterval) =
-      (epochAfterProp + minWait <= epochNo) &&
-      (case mExpected of
-        Nothing -> True
-        Just expected -> epochInterval == expected) &&
-      (epochNo <= epochAfterProp + maxWait)
+    isSuccess :: (HasCallStack, MonadTest m)
+              => AnyNewEpochState -> m (Maybe ())
+    isSuccess (AnyNewEpochState sbe newEpochState) =
+      caseShelleyToBabbageOrConwayEraOnwards
+        (const $ error "activityChangeProposalTest: Only conway era onwards supported")
+        (const $ do
+          let (EpochInterval epochInterval) = newEpochState ^. nesEpochStateL . epochStateGovStateL . curPParamsGovStateL . ppDRepActivityL
+          return (case mExpected of
+                    Nothing -> Just ()
+                    Just expected -> if epochInterval == expected then Just () else Nothing)
```

Comment:
So this means that in case of `mExpected == Nothing` we're basically skipping the check? If that's the case, can you don't use `mExpected` here, and pass an `EpochInterval` not wrapped in maybe as a function argument? You'd need to make the check before calling `watchEpochStateView`.

Reply:
Fixed in: https://github.com/IntersectMBO/cardano-node/pull/5830/commits/228b1a87d347ef7bd4237b48f87ff7532475725d